app/src/main/java/org/eyeseetea/malariacare/layout/adapters/survey/DynamicTabAdapter.java

```diff
@@ -264,6 +263,44 @@ public void OnOptionAnswered(View view, OptionDB selectedOptionDB, boolean moveT
         } else if (!(view instanceof ImageRadioButtonSingleQuestionView)) {
             showConfirmCounter(view, selectedOptionDB, questionDB, counterQuestionDB);
         }
+        if (selectedOptionDB.getId_option() != 0) {
+            moveFocusNextQuestion(view);
+        }
+    }
+
+    private void moveFocusNextQuestion(View currentView) {
```

Comment:
@manuelplazaspalacio there's already a moveFocusToNext() method in AKeyboardQuestionView that shares some of this logic. Can you please give a look at this and try to share as much logic as possible? The result I would leave it into CommonQuestionView when possible, to try to keep DynamicTabAdapter with a more reasonable size.